### PR TITLE
refactor(ssm): API cleanups

### PR DESCRIPTION
**Pull request by eladb**
BREAKING CHANGE: `ParameterStoreString` has been removed. Use `StringParameter.fromStringParameterAttributes`.
* **ssm:** `ParameterStoreSecureString` has been removed. Use `StringParameter.fromSecureStringParameterAttributes`.
* **ssm:** `ParameterOptions.name` was renamed to `parameterName`.

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [x] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-ssm/lib/parameter.ts [hunk 1]
```diff
@@ -145,8 +145,35 @@ abstract class ParameterBase extends Resource implements IParameter {
 }
 
 const STRING_PARAM_TYPE = 'String';
+const SECURE_STRING_PARAM_TYPE = 'SecureString';
 const STRINGLIST_PARAM_TYPE = 'StringList';
 
+export interface ParameterAttributes {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ssm/lib/parameter.ts]: Should be called `StringParameterAttributes` to be consistent with `SecureStringParameterAttributes`.

File: packages/@aws-cdk/aws-ssm/test/test.parameter.ts [hunk 2]
```diff
@@ -128,12 +129,15 @@ export = {
...
   },
 
-  'StringParameter.fromName'(test: Test) {
+  'StringParameter.fromStringParameterAttributes'(test: Test) {
     // GIVEN
     const stack = new Stack();
 
     // WHEN
-    const param = ssm.StringParameter.fromStringParameterName(stack, 'MyParamName', 'MyParamName');
+    const param = ssm.StringParameter.fromStringParameterAttributes(stack, 'MyParamName', {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ssm/test/test.parameter.ts]: Ehrmagerd the length of this function call :(

File: packages/@aws-cdk/aws-ssm/lib/parameter.ts (hunk 1 above)
Comment (eladb) [reply packages/@aws-cdk/aws-ssm/lib/parameter.ts]: done

File: packages/@aws-cdk/aws-ssm/test/test.parameter.ts (hunk 2 above)
Comment (eladb) [reply packages/@aws-cdk/aws-ssm/test/test.parameter.ts]: yak I know

**Pull request merged**